ext/openssl/openssl.c

```diff
@@ -879,9 +886,9 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 			php_openssl_store_errors();
 		}
 
-		if (needs_free) {
+		if (to_add_buf != NULL) {
 			/* ASN1_STRING_to_UTF8(3): The buffer out should be freed using free(3) */
-			OPENSSL_free(to_add);
+			OPENSSL_free((char *)to_add);
```

Comment:
just a small detail - it would be a little bit more cleaner to free `to_add_buf` without the cast